dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/security/UserAccountController.java

```diff
@@ -0,0 +1,158 @@
+import org.hisp.dhis.common.IllegalQueryException;
+import org.hisp.dhis.common.OpenApi;
+import org.hisp.dhis.feedback.BadRequestException;
+import org.hisp.dhis.feedback.ConflictException;
+import org.hisp.dhis.feedback.ErrorCode;
+import org.hisp.dhis.feedback.ForbiddenException;
+import org.hisp.dhis.feedback.NotFoundException;
+import org.hisp.dhis.setting.SystemSettingManager;
+import org.hisp.dhis.user.CredentialsInfo;
+import org.hisp.dhis.user.PasswordValidationResult;
+import org.hisp.dhis.user.PasswordValidationService;
+import org.hisp.dhis.user.RestoreOptions;
+import org.hisp.dhis.user.RestoreType;
+import org.hisp.dhis.user.User;
+import org.hisp.dhis.user.UserService;
+import org.hisp.dhis.webapi.mvc.annotation.ApiVersion;
+import org.hisp.dhis.webapi.utils.ContextUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author Morten Svanæs <msvanaes@dhis2.org>
+ */
+@OpenApi.Tags({"user", "login", "account"})
+@RestController
+@RequestMapping("/auth")
+@RequiredArgsConstructor
+@ApiVersion({DhisApiVersion.DEFAULT, DhisApiVersion.ALL})
+@Slf4j
+public class UserAccountController {
+
+  private final UserService userService;
+  private final SystemSettingManager systemSettingManager;
+  private final PasswordValidationService passwordValidationService;
+
+  @PostMapping("/forgotPassword")
+  @ResponseStatus(HttpStatus.OK)
+  public void forgotPassword(
+      HttpServletRequest request, @RequestBody ForgotPasswordRequest forgotPasswordRequest)
+      throws NotFoundException, ConflictException, ForbiddenException {
+
+    String username = forgotPasswordRequest.getUsername();
+
+    if (userService.isRecoveryLocked(username)) {
+      throw new ForbiddenException(
+          "The account recovery operation for the given user is temporarily locked due to too "
+              + "many calls to this endpoint in the last '"
+              + RECOVERY_LOCKOUT_MINS
+              + "' minutes. Username:"
+              + username);
+    } else {
+      userService.registerRecoveryAttempt(username);
+    }
+
+    User user = userService.getUserByUsername(username);
+
+    if (user == null) {
+      throw new NotFoundException("User does not exist: " + username);
+    }
+
+    ErrorCode errorCode = userService.validateRestore(user);
+
+    if (errorCode != null) {
+      throw new IllegalQueryException(errorCode);
+    }
+
+    if (!userService.sendRestoreOrInviteMessage(
+        user, ContextUtils.getContextPath(request), RestoreOptions.RECOVER_PASSWORD_OPTION)) {
+      throw new ConflictException("Account could not be recovered");
+    }
+
+    log.info("Recovery message sent for user: " + username);
```

Comment:
Use `{}` syntax and variable replacement for Slf4j instead of string concatenation.